demos/smart_framing_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,287 @@
+                    }
+                }
+                if (isGoodResult) {
+                    objects.push_back(obj1);
+                }
+            }
+        }
+        else {
+            // Classic postprocessing
+            std::sort(initial_objects.begin(), initial_objects.end(), [](const custom::DetectedObject& x, const custom::DetectedObject& y) { return x.confidence > y.confidence; });
+            for (size_t i = 0; i < initial_objects.size(); ++i) {
+                if (initial_objects[i].confidence == 0)
+                    continue;
+                for (size_t j = i + 1; j < initial_objects.size(); ++j)
+                    if (post_processor.intersectionOverUnion(initial_objects[i], initial_objects[j]) >= boxIOUThreshold)
+                        initial_objects[j].confidence = 0;
+                objects.push_back(initial_objects[i]);
+            }
+        }
+        slog::debug << "Filtered DetectedObject size " << objects.size() << slog::endl;
+    }
+};
+
+GAPI_OCV_KERNEL(OCVSmartFramingKernel, custom::GSmartFramingKernel) {
+    static void run(const cv::Mat & image, const std::vector<custom::DetectedObject> &objects, cv::Mat & out) {
+        cv::Rect init_rect;
+        for (const auto& el : objects) {
+            if (el.labelID == 0) {//person ID
+                init_rect = init_rect | (cv::Rect)el;
+            }
+        }
+        slog::debug << "SF result rect" << init_rect << slog::endl;
+        if (!init_rect.empty()) {
+            cv::Rect even_rect;
+            even_rect.x = (init_rect.x % 2 == 0) ? (init_rect.x) : (init_rect.x - 1);
+            even_rect.y = (init_rect.y % 2 == 0) ? (init_rect.y) : (init_rect.y - 1);
+            even_rect.width = (init_rect.width % 2 == 0) ? (init_rect.width) : (init_rect.width - 1);
+            even_rect.height = (init_rect.height % 2 == 0) ? (init_rect.height) : (init_rect.height - 1);
+
+            cv::Mat SF_ROI;
+            image(even_rect).copyTo(SF_ROI);
+
+            cv::Mat SF_resized_ROI;
+            cv::Size target_size;
+            target_size.height = image.size().height;
+            target_size.width = even_rect.width * (static_cast<float>(image.size().height) / static_cast<float>(even_rect.height));
+            target_size.width = (target_size.width % 2 == 0) ? (target_size.width) : (target_size.width - 1);
+            if (target_size.width > image.size().width) {
+                target_size.width = image.size().width;
+            }
+            cv::resize(SF_ROI, SF_resized_ROI, target_size);
+            int left_add = (image.size().width - target_size.width) / 2;
+            int right_add = (image.size().width - target_size.width) / 2;
+            cv::copyMakeBorder(/*croped_rect*/ SF_resized_ROI, out,
+                0,
+                0,
+                left_add,
+                right_add,
+                cv::BORDER_CONSTANT, cv::Scalar::all(127));
+        }
+        else {
+            image.copyTo(out);
```

Comment:
Is this copy necessary? You can get `image` for next operations from previous graph stage.

Reply:
yes. we need this copy.

Reply:
Why don't you use frame from cv::gapi::copy() instead?